lib/libutee/include/pta_system.h

```diff
@@ -160,4 +160,22 @@
  */
 #define PTA_SYSTEM_REMAP		9
 
+/*
+ * (libdl) dlopen(const char *filename, int flags)
+ *
+ * [in]     memref[0]: the UUID of the shared library to open (@filename) or
+ *                     the nil UUID (all zeros) when @filename == NULL
```

Comment:
` or the nil UUID (all zeros) when @filename == NULL` relates to `dlsym`, not `dlopen`.

Reply:
Well it relates to both actually.

Reply:
ok, fine as is.

Reply:
Now I realize there is a small mistake in my handling of NULL in `dlopen()` and `dlsym()`, and in my test case too. I am doing `dlsym(NULL, ...)` which is not valid ; when one wants to open the main executable, one should call `h = dlopen(NULL, ...); dlsym(h, ...);`. I'll fix and update the comments too.

Reply:
From `man dlsym`:
```
       The  function  dlsym()  takes  a "handle" of a dynamic loaded shared
       object returned by dlopen(3) along with a null-terminated symbol name,
       and (...)
```
From `man dlopen`:
```
RETURN VALUE
       On  success,  dlopen() and dlmopen() return a non-NULL handle for
       the loaded library.  On error (file could not be found, was not
       readable, had the wrong format, or caused errors during loading),
       these functions return NULL.
       (...)
       Errors from these functions can be diagnosed using dlerror(3).
```
Should one the value returned from `dlopen()` is safe, calling `dlerror()`, before calling `dlsym()`? It this operation really needed from the PTA?

Reply:
Sorry, i mixed-up those too APIs to. Indeed `dlopen(NULL,` ...) returns a valid non null handle.
You can discard/resolve this comment thread about the inline description.